torch/_inductor/codegen/triton.py

```diff
@@ -1703,6 +1708,71 @@ def call_kernel(self, code, name: str):
             V.graph.scheduler.current_device.index,
         )
 
+    def warn_mix_layout(self, kernel_name):
```

Comment:
This is only for debug purposes?

Reply:
yea. it's disabled by default but can be enabled when needed. I feel it can be useful to keep it. e.g., I use this to find the root cause for the slowness of yolov3 previously due to mix layout inputs and add the second pass in `GraphLowering.find_nodes_prefer_channels_last` to fix that. The fix improves quite a few other models as well. (Check comment in that method for more details).